### ISSUE TITLE
Tunneling Proxy doesn't work if the hooked origin has the https scheme

### ISSUE THREAD

**Issue by am06** [beefproject/beef#764]
Hello,

i have hooked a browser via persistent xss, set the zombie as proxy, withe another browser i've conected to 127.0.0.1:6789, i try to browser to my target website but it doesn't work.
Also a lot of other modules (as green) do not work.

Thanks.

**Comment by bcoles** [comment]
This should now be resolved along with several other bugs in the requester and proxy.

* The requester `Http` model was modified to add a `proto` property which specifies the URL scheme for the request in 95abdf6781323902745f49cf5aa530dd2659ce28.
* The requester was modified to make use of the `proto` property to specify whether the request should make use of SSL in 95abdf6781323902745f49cf5aa530dd2659ce28.
* The `proto` property was added to the requester history in 10d79b9a715044010659b9236bc0d80a7293aacb.
* A configurable checkbox was added to the 'Forge Request' tab in the UI to specify whether the request should make use of SSL in 10d79b9a715044010659b9236bc0d80a7293aacb.
* The proxy request parsing method was modified to make use of the `proto` property to specify whether the request should make use of SSL in 08f5cf3e29e884bed8f78a4c42b2e65e175357d8.
* The `beef.net.forge_request` client-side method was modified to fail gracefully in the event a request attempts to load a HTTP resource from a HTTPS origin - returning `mixedcontent` - in 91bfb06f8cf0ae4d52c159b476e4d950c2beb3cc.

Tested to HTTP and HTTPS from HTTP and HTTPS origins on IE8, IE11, Safari 5, Firefox and Chrome.

:hammer: :bug:

No functionality has been added to delete a request from the history. I've added an issue to the tracker here: #1388.

Closing this issue.
